tests/e2e/framework/kubernetes.go

```diff
@@ -491,12 +491,20 @@ func (k *KubeInfo) Teardown() error {
 	// confirm the namespace is deleted as it will cause future creation to fail
 	maxAttempts := 600
 	namespaceDeleted := false
+	validatingWebhookConfigurationDeleted := false
 	log.Infof("Deleting namespace %v", k.Namespace)
 	for attempts := 1; attempts <= maxAttempts; attempts++ {
 		namespaceDeleted, _ = util.NamespaceDeleted(k.Namespace, k.KubeConfig)
-		if namespaceDeleted {
+		// As validatingWebhookConfiguration "istio-galley" will
+		// be delete by kubernetes GC controller asynchronously,
+		// we need to ensure it's deleted before return.
+		// TODO: find a more general way as long term solution.
+		validatingWebhookConfigurationDeleted = util.ValidatingWebhookConfigurationDeleted("istio-galley", k.KubeConfig)
```

Comment:
is the name is always "istio-galley"? If so, do we even need to parameterize it in util.ValidatingWebhookConfigurationDeleted?

Reply:
My understanding is what `istio.io/istio/tests/util` is fairly generic k8s test utility code. The configuration validation webhook is typically named `istio-galley`, but in theory istio-galley could be renamed and/or we find other uses for validating webhooks.